csv/generic/shared/src/test/scala-2/fs2/data/csv/generic/CsvRowDecoderDefaultsTest.scala

```diff
@@ -0,0 +1,34 @@
+package fs2.data.csv.generic
+
+import cats.data.NonEmptyList
+import fs2.data.csv.CsvRow
+import fs2.data.csv.generic.semiauto.deriveCsvRowDecoder
+import weaver.SimpleIOSuite
+
+object CsvRowDecoderDefaultsTest extends SimpleIOSuite {
+
+  val csvRowDefaultI = CsvRow.unsafe(NonEmptyList.of("", "test", "42"), NonEmptyList.of("i", "s", "j"))
+  val csvRowNoI =
+    CsvRow.unsafe(NonEmptyList.of("test", "42"), NonEmptyList.of("s", "j"))
+  val csvRowEmptyJ =
+    CsvRow.unsafe(NonEmptyList.of("1", "test", ""), NonEmptyList.of("i", "s", "j"))
+
+  case class Test(i: Int = 0, s: String, j: Option[Int])
+  case class TestDefaults(i: Int = 7, j: String = "foo")
+
+  val testDecoder = deriveCsvRowDecoder[Test]
+  val testDefaultsDecoder = deriveCsvRowDecoder[TestDefaults]
+
+  pureTest("case classes should be handled properly with default value and empty cell") {
+    expect(testDecoder(csvRowDefaultI) == Right(Test(0, "test", Some(42))))
+  }
+
+  pureTest("case classes should be handled properly with default value and missing column") {
+    expect(testDecoder(csvRowNoI) == Right(Test(0, "test", Some(42))))
+  }
+
+  pureTest("case classes should be handled properly with default string value and empty cell") {
+    expect(testDefaultsDecoder(csvRowEmptyJ) == Right(TestDefaults(1, "foo")))
```

Comment:
Note that the default `"foo"` is expected even though a `String` cell could handle the empty cell. While I think it's the better behaviour for users as they now can handle `Int` or any other type that can't be parsed from empty input via default, it is a change that we should highlight in the release notes. Users that need to handle empty cells may no longer define a default (except for `""` itself) for those in their `case class`.

Reply:
Thinking about it, I would add a clear note in the documentation for this caveat.

Reply:
Done in ad960a3f20d869f234dd349b4cc016077322a6cc